Libraries/xcexecution/Sources/SimpleExecutor.cpp

```diff
@@ -199,8 +200,15 @@ writeAuxiliaryFiles(
                 xcformatter::Formatter::Print(_formatter->setAuxiliaryExecutable(auxiliaryFile.path()));
 
                 if (!_dryRun) {
-                    // FIXME: This should use the filesystem.
-                    if (::chmod(auxiliaryFile.path().c_str(), S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) != 0) {
+                    Permissions permissions;
```

Comment:
lol. yes it is pretty bad :(

Maybe could offer something like this:

    Permissions(
        Permissions::All,
        { Permissions::Read, Permissions::Execute },
        { Permissions::Read, Permissions::Execute })

With `std::initializer_list` or something parameters. Cuts it down to three lines versus up to 10?